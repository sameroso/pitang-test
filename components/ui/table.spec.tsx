import { render, screen } from "@testing-library/react";
import {
  Table,
  TableBody,
  TableCaption,
  TableCell,
  TableFooter,
  TableHead,
  TableHeader,
  TableRow,
} from "./table";

describe("Table tests", () => {
  it("should render table correctly", () => {
    render(<Table />);
    expect(screen.getByRole("table")).toBeInTheDocument();
  });

  it("should table match snapshot", () => {
    const { container } = render(<Table />);
    expect(container).toMatchSnapshot();
  });
});

describe("Table header tests", () => {
  it("should render table header", () => {
    const { container } = render(<TableHeader />);
    expect(container).toBeInTheDocument();
  });

  it("should table header match snapshot", () => {
    const { container } = render(<TableHeader />);
    expect(container).toMatchSnapshot();
  });
});

describe("Table body tests", () => {
  it("should render table body", () => {
    const { container } = render(<TableBody />);
    expect(container).toBeInTheDocument();
  });

  it("should table body match snapshot", () => {
    const { container } = render(<TableBody />);
    expect(container).toMatchSnapshot();
  });
});

describe("Table footer tests", () => {
  it("should render table footer", () => {
    const { container } = render(<TableFooter />);
    expect(container).toBeInTheDocument();
  });

  it("should table footer match snapshot", () => {
    const { container } = render(<TableFooter />);
    expect(container).toMatchSnapshot();
  });
});

describe("Table row tests", () => {
  it("should render table row", () => {
    const { container } = render(<TableRow />);
    expect(container).toBeInTheDocument();
  });

  it("should table row match snapshot", () => {
    const { container } = render(<TableRow />);
    expect(container).toMatchSnapshot();
  });
});

describe("Table head tests", () => {
  it("should render table head", () => {
    const { container } = render(<TableHead />);
    expect(container).toBeInTheDocument();
  });

  it("should table head match snapshot", () => {
    const { container } = render(<TableHead />);
    expect(container).toMatchSnapshot();
  });
});

describe("Table cell tests", () => {
  it("should render table cell", () => {
    const { container } = render(<TableCell />);
    expect(container).toBeInTheDocument();
  });

  it("should table cell match snapshot", () => {
    const { container } = render(<TableCell />);
    expect(container).toMatchSnapshot();
  });
});

describe("Table caption tests", () => {
  it("should render table caption", () => {
    const { container } = render(<TableCaption>caption</TableCaption>);
    expect(container).toBeInTheDocument();
  });

  it("should table caption match snapshot", () => {
    const { container } = render(<TableCaption>caption</TableCaption>);
    expect(container).toMatchSnapshot();
  });
});
