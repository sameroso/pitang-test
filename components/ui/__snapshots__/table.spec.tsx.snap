// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table body tests should table body match snapshot 1`] = `
<div>
  <tbody
    class="[&_tr:last-child]:border-0"
  />
</div>
`;

exports[`Table caption tests should table caption match snapshot 1`] = `
<div>
  <caption
    class="mt-4 text-sm text-muted-foreground"
  >
    caption
  </caption>
</div>
`;

exports[`Table cell tests should table cell match snapshot 1`] = `
<div>
  <td
    class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
  />
</div>
`;

exports[`Table footer tests should table footer match snapshot 1`] = `
<div>
  <tfoot
    class="border-t bg-muted/50 font-medium [&>tr]:last:border-b-0"
  />
</div>
`;

exports[`Table head tests should table head match snapshot 1`] = `
<div>
  <th
    class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
  />
</div>
`;

exports[`Table header tests should table header match snapshot 1`] = `
<div>
  <thead
    class="[&_tr]:border-b"
  />
</div>
`;

exports[`Table row tests should table row match snapshot 1`] = `
<div>
  <tr
    class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
  />
</div>
`;

exports[`Table tests should table match snapshot 1`] = `
<div>
  <div
    class="relative w-full overflow-auto"
  >
    <table
      class="w-full caption-bottom text-sm"
    />
  </div>
</div>
`;
